src/Microsoft.Graph.Core/Models/UploadSession.cs

```diff
@@ -4,8 +4,10 @@
 
 namespace Microsoft.Graph.Core.Models
 {
+    using Microsoft.Kiota.Abstractions.Serialization;
```

Comment:
I'm curious to what's the plan for this "handcrafted" upload session vs the code-gen ones?
The customer will have access to something like `client.Me.Drive.Items["id"].Value.CreateUploadSession.PostAsync()`, which will return an upload session from the code-gen, looking a lot like this one, but not implementing the interface.
Now the large file upload task itself takes an IUploadSession, which is implemented by this class. How do you plan people going from what the fluent API returned to this?

Reply:
This would ideally be achieved through extending the model class in the service library. In my head we could achieve this using partial classes as we already do currently. 
The alternative would be to use your suggestion of a config file, which would simply enrich matching classes with relevant code element properties (in this case implementing the interface).

Reply:
partial classes are a very dotnet approach to this issue, I'd like to address this aspect in a similar way across languages if we can to ease up the maintenance burden on language owner. I'll go ahead and capture what we said in the issue you linked. For now we can keep this duplicate class, I'll go ahead and approve this PR, knowing we'll have to finalize the work once the extensibility story is tackled in kiota.